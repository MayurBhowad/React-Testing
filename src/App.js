import logo from './logo.svg';
import './App.css';
import Button from './components/button/Button';
import { useState } from 'react';

function App() {
  const [buttonClicked, setButtonClicked] = useState(false);
  const fetching = async () => {
    setButtonClicked(true);
  }

  const buttonText = buttonClicked ? 'Ok' : 'fetch';
  return (
    <div className="App">
      <header className="App-header">
        <Button lable="Click me please"></Button>
      </header>
      <a href="">React Testing</a>
      <button className="btn btn-success" onClick={fetching} disabled={buttonClicked}>{buttonText}</button>
    </div>
  );
}

export default App;
