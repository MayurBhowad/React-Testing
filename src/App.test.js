import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import App from './App';

it('should take snapSHot', () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
})