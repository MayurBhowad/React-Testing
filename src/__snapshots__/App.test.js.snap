// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take snapSHot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="button-style"
        data-testid="button"
      >
        Hi Click me please
      </div>
    </header>
    <a
      href=""
    >
      Learn React
    </a>
    <button
      class="btn btn-success"
    >
      fetch
    </button>
  </div>
</DocumentFragment>
`;
